### ISSUE TITLE
undefined method `session_options' for #<Capybara::Session>

### ISSUE THREAD

**Issue by johnmeehan** [teamcapybara/capybara#1883]
## Meta
Capybara Version:
2.14.3
Driver Information (and browser if relevant):
Poltergeist
RSpec Feature test
ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin16]
Rails 5.0.3

## Actual Behavior

Running RSpec I see a no method error for session_options in Capybara.

The Green! test:
```ruby
  # /spec/features/frontend/company/engagement/new_advisory_spec.rb
  feature "form field validations" do
    before { click_link 'Next' }
    subject { page }
    it { is_expected.to have_content 'Please select an office.'}
    it { is_expected.to have_content 'Please select at least one topic.'}
    it { is_expected.to have_selector 'input#date+span',text: 'This is a required field'}
    it { is_expected.to_not have_button("Next")}
  end
```

Formatted test suite output with capybara error: (all green)
```
Company user making a new engagement request
  form field validations
    example at ./spec/features/frontend/company/engagement/new_advisory_spec.rb:107 (Got an error when generating description from matcher: NoMethodError: undefined method `session_options' for #<Capybara::Session> -- /Users/john/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/capybara-2.14.3/lib/capybara/rspec/matchers.rb:48:in `session_options')
    should text "Please select at least one topic."
    example at ./spec/features/frontend/company/engagement/new_advisory_spec.rb:108 (Got an error when generating description from matcher: NoMethodError: undefined method `session_options' for #<Capybara::Session> -- /Users/john/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/capybara-2.14.3/lib/capybara/rspec/matchers.rb:48:in `session_options')
    should text "Please select an office."
```

## capybara-2.14.3/lib/capybara/rspec/matchers.rb
```
[43, 52] in /Users/john/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/capybara-2.14.3/lib/capybara/rspec/matchers.rb
   43:         @args
   44:       end
   45: 
   46:       def session_options
   47:         byebug
=> 48:         @context_el ||= nil
   49:         @context_el ? @context_el.session_options : Capybara.session_options
   50:       end
   51:     end
   52: 
(byebug) Capybara.session_options
#<Capybara::SessionConfig:0x007fb622c87230 @server_host=nil, @always_include_port=true, @run_server=true, @default_selector=:css, @default_max_wait_time=15, @ignore_hidden_elements=true, @default_host="http://www.example.com", @automatic_reload=true, @match=:smart, @exact=false, @exact_text=false, @raise_server_errors=true, @server_errors=[StandardError], @visible_text_only=false, @wait_on_first_by_default=false, @automatic_label_click=false, @enable_aria_label=false, @save_path=#<Pathname:/Users/john/Documents/meetyl/tmp/capybara>>
(byebug) @context_el
#<Capybara::Session>
(byebug) @context_el.session_options
*** NoMethodError Exception: undefined method `session_options' for #<Capybara::Session>
```

**Comment by twalpole** [comment]
Please post the full stacktrace so I can see how this is happening.  Also, note that the way you're writing feature tests is not good.  You're writing them like unit tests, which means for every `it` the "browser" has to fully reload the page, parse the JS, etc, etc to check one string of text.  A feature test (as the name implies) is meant to test a whole feature, not one line of text..

**Comment by twalpole** [comment]
hmmm -- I see -- it's when the have_selector matcher fails

**Comment by twalpole** [comment]
Please try the master branch, should be fixed via 442c8d29770c4e40a1f5a5f1e9c2f477b8db36c5 and you'll now get the correct error messages for your expectation failures

**Comment by twalpole** [comment]
Fix pushed in 2.14.4

**Comment by johnmeehan** [comment]
Awesome! 2.14.4 fixes the issue.
Of course my spec code was not entirely helpful to begin with.

Thanks for the quick fix @twalpole
